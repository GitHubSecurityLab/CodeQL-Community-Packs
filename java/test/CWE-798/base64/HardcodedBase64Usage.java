import java.util.Base64;

class MyApp {
    public static String KEY = "VGVzdFBhc3N3b3Jk";
    public static byte[] KEY2 = new byte[] { 'V', 'G', 'V', 'z', 'd', 'F', 'B', 'h', 'c', '3', 'N', '3', 'b', '3', 'J',
            'k' };

    public String getDecoderString() {
        byte[] decodedBytes = Base64.getDecoder().decode(MyApp.KEY);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderBytes() {
        byte[] decodedBytes = Base64.getDecoder().decode(MyApp.KEY2);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderConvertString() {
        String key = new String(MyApp.KEY2);
        byte[] decodedBytes = Base64.getDecoder().decode(key);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderConvertBytes() {
        byte[] key = MyApp.KEY.getBytes();
        byte[] decodedBytes = Base64.getDecoder().decode(key);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String simpleDecoderString() {
        byte[] decodedBytes = Base64.getDecoder().decode("U2VjcmV0S2V5");

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

}