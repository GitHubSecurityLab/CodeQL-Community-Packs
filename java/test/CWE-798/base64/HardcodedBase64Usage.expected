sources
| HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" |
| HardcodedBase64Usage.java:5:33:6:17 | new byte[] |
| HardcodedBase64Usage.java:39:58:39:71 | "U2VjcmV0S2V5" |
sinks
| HardcodedBase64Usage.java:9:58:9:66 | MyApp.KEY |
| HardcodedBase64Usage.java:16:58:16:67 | MyApp.KEY2 |
| HardcodedBase64Usage.java:24:58:24:60 | key |
| HardcodedBase64Usage.java:32:58:32:60 | key |
| HardcodedBase64Usage.java:39:58:39:71 | "U2VjcmV0S2V5" |
